src/utilities/util_verify.c

```diff
@@ -122,14 +133,34 @@ util_verify(WT_SESSION *session, int argc, char *argv[])
             goto err;
         }
     }
-    if ((ret = session->verify(session, uri, config)) != 0)
-        (void)util_err(session, ret, "session.verify: %s", uri);
-    else {
-        /*
-         * Verbose configures a progress counter, move to the next line.
-         */
-        if (verbose)
-            printf("\n");
+
+    /* Verify all the tables if no particular URI is specified. */
+    if (argc < 1) {
+        /* Open the metadata file and iterate through its entries, verifying each one. */
+        if ((ret = session->open_cursor(session, WT_METADATA_URI, NULL, NULL, &cursor)) != 0) {
+            /*
+             * If there is no metadata (yet), this will return ENOENT. Treat that the same as an
+             * empty metadata.
+             */
+            if (ret == ENOENT)
+                ret = 0;
+
+            WT_ERR(util_err(session, ret, "%s: WT_SESSION.open_cursor", WT_METADATA_URI));
+        }
+
+        while ((ret = cursor->next(cursor)) == 0) {
```

Comment:
What should `util_verify()` return when run on all tables? It looks like it will always return an error value (WT_NOTFOUND?) from the last call to `cursor->next()`. This probably isn't what we want.

Reply:
Thanks for pointing that out, I didn't implement this correctly at all. After thinking about it I'd like it to still return an error if verify fails on one of the tables but I do think it should keep going and try to verify the rest of the tables.

Reply:
It could hide an actual error in the `cursor.next` implementation though but I cannot see any other way to avoid this as of now unless we know how many entries we are supposed to go through.

Reply:
> It could hide an actual error in the cursor.next implementation

You could explicitly check for the expected error -- `WT_NOTFOUND` or whatever gets returned when we hit the end of the table, and if there is some other error return it.

But the primary goal here is to verify the user's data. This isn't a correctness test for cursors. So IMO we need to worry about this case too much.

Reply:
> Thanks for pointing that out, I didn't implement this correctly at all. After thinking about it I'd like it to still return an error if verify fails on one of the tables but I do think it should keep going and try to verify the rest of the tables.

No worries. Everyone has PRs that don't work on the first go.  

I agree that it is fine to keep going and verify all the tables. You'll just have to decide whether to return the first or last error. It probably doesn't matter as long as they all get reported in the output.

It might be interesting to have an option where the user can say whether they want to stop on the 1st error. But that seems like a separate, additional ticket and not something that should happen here.